src/Microsoft.Azure.WebJobs.Host/Bindings/BindingProviders/FilteringBindingProvider.cs

```diff
@@ -26,7 +33,16 @@ public Task<IBinding> TryCreateAsync(BindingProviderContext context)
                 throw new ArgumentNullException("context");
             }
             Type parametertype = context.Parameter.ParameterType;
-            if (!_predicate(parametertype))
+
+            var attr = context.Parameter.GetCustomAttribute<TAttribute>();
+
+            var cloner = new AttributeCloner<TAttribute>(attr, _nameResolver);
+            var attrNameResolved = cloner.GetNameResolvedAttribute();
+
+            // This may do validation and throw too. 
+            bool canBind = _predicate(attrNameResolved, parametertype);
+
+            if (!canBind)
```

Comment:
Do AddFilter( t => IsTypeA(t) || IsTypeB(t) || IsTypeC(t), ...) 

---
In reply to: [63770416](https://github.com/Azure/azure-webjobs-sdk/pull/726#discussion_r63770416) [](ancestors = 63770416)